python/ray/data/_internal/arrow_block.py

```diff
@@ -70,6 +71,19 @@ class ArrowRow(TableRow):
     """
 
     def __getitem__(self, key: str) -> Any:
+        from ray.data.extensions.tensor_extension import (
+            ArrowTensorType,
+            ArrowVariableShapedTensorType,
+        )
+
+        schema = self._row.schema
+        if isinstance(
+            schema.field(schema.get_field_index(key)).type,
```

Comment:
nit: can it just be `schema.field(key).type` - https://arrow.apache.org/docs/python/generated/pyarrow.Schema.html#pyarrow.Schema.field ?

Reply:
Hmm I had thought that this support was added in a post-Arrow 6 version, but looks like it's supported. I'll make this change!